backend/api/blueprints/leads.py

```diff
@@ -361,6 +365,45 @@ def _create_new_lead(request, valid_data_sources=VALID_DATA_SOURCES):
             if not all(c.isalnum() or c.isspace() or c == "." or c == "&" or c == "'" for c in company_name):
                 return {"message": "company_name can only contain a-z, 0-9, ., &, ', whitespace"}, 400
 
+    # using pydantic model to check phone number, assigned, email and social media urls
+    class FieldTestModel(BaseModel):
+        phoneNum: str
+        assigned: str
+        email: EmailStr
+        socialLink: HttpUrl
+
+        @validator("phoneNum")
+        def check_phoneNumber_format(cls, v):
+            regExs = (r"\(\w{3}\) \w{3}\-\w{4}", r"^\w{3}\-\w{4}$")
+            if not re.search(regExs[0], v):
+                return ValueError("not match")
+            return v
+
+        @validator("assigned")
+        def check_assigned(cls, v):
+            users_response = get_all_users()
```

Comment:
We shouldn't need to load all users into memory to check this. We should consider using `_get_user_by_username` here or writing a sql query that counts the number of users with a given username.